File: student_info_system/android/app/src/main/java/com/shadedgames/student_info_system/helper_classes/Grades.java
package com.shadedgames.student_info_system.helper_classes;

import com.google.gson.Gson;

import java.util.HashMap;
import java.util.Map;

public class Grades {

    private int id;
    private String stdId;
    private double firstSub;
    private double secondSub;
    private double thirdSub;
    private double fourthSub;
    private double fifthSub;
    private double sixthSub;
    private double seventhSub;
    private double eightSub;
    private double ninthSub;

    public Grades() {

    }

    // constructor from a Database INPUT
    public Grades(int id, String stdId, double firstSub, double secondSub, double thirdSub, double fourthSub, double fifthSub, double sixthSub, double seventhSub, double eightSub, double ninthSub) {
        setId(id);
        setStdId(stdId);
        setFirstSub(firstSub);
        setSecondSub(secondSub);
        setThirdSub(thirdSub);
        setFourthSub(fourthSub);
        setFifthSub(fifthSub);
        setSixthSub(sixthSub);
        setSeventhSub(seventhSub);
        setEightSub(eightSub);
        setNinthSub(ninthSub);
    }

    // constructor from a JSON STRING INPUT
    public Grades(String stdId, double firstSub, double secondSub, double thirdSub, double fourthSub, double fifthSub, double sixthSub, double seventhSub, double eightSub, double ninthSub) {
        setStdId(stdId);
        setFirstSub(firstSub);
        setSecondSub(secondSub);
        setThirdSub(thirdSub);
        setFourthSub(fourthSub);
        setFifthSub(fifthSub);
        setSixthSub(sixthSub);
        setSeventhSub(seventhSub);
        setEightSub(eightSub);
        setNinthSub(ninthSub);
    }

    public double getEightSub() {
        return eightSub;
    }

    public double getFifthSub() {
        return fifthSub;
    }

    public double getFirstSub() {
        return firstSub;
    }

    public double getFourthSub() {
        return fourthSub;
    }

    public double getNinthSub() {
        return ninthSub;
    }

    public double getSecondSub() {
        return secondSub;
    }

    public double getSeventhSub() {
        return seventhSub;
    }

    public double getSixthSub() {
        return sixthSub;
    }

    public double getThirdSub() {
        return thirdSub;
    }

    public int getId() {
        return id;
    }

    public String getStdId() {
        return stdId;
    }

    public void setEightSub(double eightSub) {
        this.eightSub = eightSub;
    }

    public void setFifthSub(double fifthSub) {
        this.fifthSub = fifthSub;
    }

    public void setFirstSub(double firstSub) {
        this.firstSub = firstSub;
    }

    public void setFourthSub(double fourthSub) {
        this.fourthSub = fourthSub;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setNinthSub(double ninthSub) {
        this.ninthSub = ninthSub;
    }

    public void setSecondSub(double secondSub) {
        this.secondSub = secondSub;
    }

    public void setSeventhSub(double seventhSub) {
        this.seventhSub = seventhSub;
    }

    public void setSixthSub(double sixthSub) {
        this.sixthSub = sixthSub;
    }

    public void setStdId(String stdId) {
        this.stdId = stdId;
    }

    public void setThirdSub(double thirdSub) {
        this.thirdSub = thirdSub;
    }

    public String testGson() {
        Gson gson = new Gson();
        Map<String, Object> profileMap = new HashMap<>();
        profileMap.put("id", getId());
        profileMap.put("stdId", getStdId());
        profileMap.put("firstSub", getFirstSub());
        profileMap.put("secondSub", getSecondSub());
        profileMap.put("thirdSub", getThirdSub());
        profileMap.put("fourthSub", getFourthSub());
        profileMap.put("fifthSub", getFifthSub());
        profileMap.put("sixthSub", getSixthSub());
        profileMap.put("seventhSub", getSeventhSub());
        profileMap.put("eightSub", getEightSub());
        profileMap.put("ninthSub", getNinthSub());
        String output = gson.toJson(profileMap);

        return output;

    }

}

File: student_info_system/lib/Models/Grades.dart
import 'package:json_annotation/json_annotation.dart';

part 'Grades.g.dart';

@JsonSerializable()
class Grades {
  @JsonKey(name: 'id')
  int id;
  @JsonKey(name: 'stdId')
  String stdId;
  @JsonKey(name: 'firstSub')
  double firstSub;
  @JsonKey(name: 'secondSub')
  double secondSub;
  @JsonKey(name: 'thirdSub')
  double thirdSub;
  @JsonKey(name: 'fourthSub')
  double fourthSub;
  @JsonKey(name: 'fifthSub')
  double fifthSub;
  @JsonKey(name: 'sixthSub')
  double sixthSub;
  @JsonKey(name: 'seventhSub')
  double seventhSub;
  @JsonKey(name: 'eightSub')
  double eightSub;
  @JsonKey(name: 'ninthSub')
  double ninthSub;

  Grades([
      this.id,
      this.stdId,
      this.firstSub,
      this.secondSub,
      this.thirdSub,
      this.fourthSub,
      this.fifthSub,
      this.sixthSub,
      this.seventhSub,
      this.eightSub,
      this.ninthSub]);

  factory Grades.fromJson(Map<String, dynamic> json) => _$GradesFromJson(
      json); // a factory method to deserialize a JSON string to a Dart Object

  Map<String, dynamic> toJson() => _$GradesToJson(this);

  get getId => id;

  get getstdId => stdId;

  get getFirstSub => firstSub;

  get getSecondSub => secondSub;

  get getThirdSub => thirdSub;

  get getFourthSub => fourthSub;

  get getFifthSub => fifthSub;

  get getSixthSub => sixthSub;

  get getSeventhSub => seventhSub;

  get getEightSub => eightSub;

  get getNinthSub => ninthSub;

  List<String> subjects()
  {
    List<String> subs = new List<String>();
    subs.add("Data Structures and Algorithms");
    subs.add("Discrete Structures");
    subs.add("Object Oriented Programming");
    subs.add("Modelling and Simulation");
    subs.add("Logic Design");
    subs.add("CS Free Elective");
    subs.add("Ethics");
    subs.add("World Literature");
    subs.add("Physical Education");
    return subs;
  }

  List<String> subjectCodes()
  {
    List<String> subs = new List<String>();
    subs.add("COMP 200337");
    subs.add("COMP 20043");
    subs.add("COMP 20083");
    subs.add("COMP 20087");
    subs.add("COMP 20039");
    subs.add("GEED 10042");
    subs.add("GEED 10093");
    subs.add("GEED 10064");
    subs.add("GEED 10022");
    return subs;
  }
  List<double> grades()
  {
    List<double> subs = new List<double>();
    subs.add(firstSub);
    subs.add(secondSub);
    subs.add(thirdSub);
    subs.add(fourthSub);
    subs.add(fifthSub);
    subs.add(sixthSub);
    subs.add(seventhSub);
    subs.add(eightSub);
    subs.add(ninthSub);


    return subs;
  }

  double average()
  {
    double g = (firstSub + secondSub + thirdSub + fourthSub +fifthSub +sixthSub + seventhSub +eightSub +ninthSub)/9;
    return g;
  }
}
